feat: Added a spec to describe the type casting we recommend to do on behalf of the user (#803)
Azure/Azure-Verified-Modules: docs/content/specs/bicep/_index.md

---

<br>

#### ID: BCPNFR17 - Category: Composition - Code Styling - Type casting

> To improve the usability of primitive module properties declared as strings, you should declare them as the type they are and apply any required casting in the module on behalf of the user.

Reviewer: Although I know what the goal is, I'm not sure if "as the type they are" clarifies. I don't have a good suggestion either, but trying: 
```suggestion
To improve the usability of primitive module properties declared as strings, you should declare them using a type which better represents them, and apply any required casting in the module on behalf of the user.
```

For reference, please refer to the following examples:

<h5>Boolean as String</h5>
{{< tabs "booleanString" >}}
  {{< tab "Before" >}}

  ```bicep
  @allowed([
    'false'
    'true'
  ])
  param myParameterValue string = 'false'

  resource myResource '(...)' = {
    (...)
    properties: {
      myParameter: myParameterValue
    }
  }
  ```

  {{< /tab >}}
  {{< tab "After" >}}

  ```bicep
  param myParameterValue string = false

  resource myResource '(...)' = {
    (...)
    properties: {
      myParameter: string(myParameterValue)
    }
  }
  ```

  {{< /tab >}}
{{< /tabs >}}

<h5>Integer Array as String Array</h5>

{{< tabs "intArrayString" >}}
  {{< tab "Before" >}}

>   ```bicep

Reviewer: Should this example change based on #817 ?
Also why `zonesValue` instead of `zones`?

Author: It should yes, @eriqua. Updated it. 

I think I called it `zoneValues` to make it explicit that I'm not refering to a property or something. Anyways, changed it to `zones`.
